File: VB/iCalendarExportImport/CustomObject.vb
Imports System
Imports System.Collections.Generic
Imports System.Text
Imports System.Drawing
Imports System.IO

Namespace iCalendarExportImport
    <Serializable> _
    Public Class CustomObject

        Private name_Renamed As String

        Private info_Renamed As String

        Private picture_Renamed() As Byte

        Public Sub New()
        End Sub
        Public Property Name() As String
            Get
                Return name_Renamed
            End Get
            Set(ByVal value As String)
                name_Renamed = value
            End Set
        End Property
        Public Property Info() As String
            Get
                Return info_Renamed
            End Get
            Set(ByVal value As String)
                info_Renamed = value
            End Set
        End Property
        Public Property PictureBytes() As Byte()
            Get
                Return picture_Renamed
            End Get
            Set(ByVal value As Byte())
                picture_Renamed = value
            End Set
        End Property
        Public Property Picture() As Image
            Get
                Dim ms As New MemoryStream(picture_Renamed)
                Return Image.FromStream(ms)

            End Get
            Set(ByVal value As Image)
                Dim ms As New MemoryStream()
                value.Save(ms, System.Drawing.Imaging.ImageFormat.Bmp)
                picture_Renamed = ms.ToArray()
            End Set
        End Property

        Public Overrides Function ToString() As String
            Return String.Format("NAME={0} INFO={1}", Name, Info)
        End Function
    End Class

End Namespace

File: CS/iCalendarExportImport/CustomObject.cs
using System;
using System.Collections.Generic;
using System.Text;
using System.Drawing;
using System.IO;

namespace iCalendarExportImport {
    [Serializable]
    public class CustomObject {
        string name;
        string info;
        byte[] picture;

        public CustomObject() {
        }
        public string Name { get { return name; } set { name = value; } }
        public string Info { get { return info; } set { info = value; } }
        public byte[] PictureBytes { get { return picture; } set { picture = value; } }
        public Image Picture {
            get {
                MemoryStream ms = new MemoryStream(picture);
                return Image.FromStream(ms); ;
            }
            set {
                MemoryStream ms = new MemoryStream();
                value.Save(ms, System.Drawing.Imaging.ImageFormat.Bmp);
                picture = ms.ToArray();
            }
        }

        public override string ToString() {
            return String.Format("NAME={0} INFO={1}", Name, Info);
        }
    }

}
